### DISCUSSION TITLE
Spotify playlists sorting issue/Play Playlist from here issue/Reverse Order playback

### DISCUSSION THREAD

**Discussion by apedance** [music-assistant/support#2252] [Feature requests and ideas]
### What version of Music Assistant has the issue?

2.0.1

### What version of the Home Assistant Integration have you got installed?

2024.5.3

### Have you tried everything in the Troubleshooting FAQ and reviewed the Open and Closed Issues and Discussions to resolve this yourself?

- [X] Yes

### The problem

I mainly use Spotify as a music source.
Most of the playlists, including my own created ones do not have the option to sort the tracks which is depicted in the picture below.
![image](https://github.com/music-assistant/support/assets/4264240/4f2de6c2-44dc-45fd-bc1f-43cbc5f2dec9)

If I load a playlist of someone else that I have subscribed to it works as seen in the picture below.
![image](https://github.com/music-assistant/support/assets/4264240/706d7803-0732-45c9-a589-a6a9cc88a17e)

My goal is to have the Position (descending) option for my own playlists.

------------------------------
This issue is sorted. Check my post below for Reverse order playback issue.

**Comment by OzGav** [comment]
You just have to give it time to load the full playlist then you should see the sort options

**Comment by apedance** [comment]
Oh, thank you. Worked. My bad.

I manually scrolled through all tracks - this made it work.

**Comment by apedance** [comment]
Okay, the problem is shifting now.
Sorting is working as seen in the picture below. (reverse oder)
![image](https://github.com/music-assistant/support/assets/4264240/76a7881e-4f20-46b5-8cb2-4a01d3e566b1)

If I click on the first song and use Play Playlist from here it will not play the songs as sorted in the gui, it will jump to song number 1 after and continue on.

Goal: I want to play the playlist in reverse order added - song 919, song 918, song 917, etc.

Thank you!

**Comment by OzGav** [comment]
Not possible. I will transfer this to a feature request

**Comment by mwdle** [comment]
I am also facing this issue, the actual playlist playback order / does not match the ordering shown on the playlist page using sorting options.

**Reply by OzGav** [reply]
That is correct. Playlists play in the order they were created. The sorting options are for viewing only. Providing the capability to reorder playlists is on the to do list.

**Reply by OzGav** [reply]
Following up on this, you can sort the playlist into the order you want, then use the use the multi select option and select all then use the actions menu that appears to add to the queue

**Comment by mdbaharzaman** [comment]
Yeah, the sorting options in Spotify playlists can be pretty limited and sometimes frustrating — especially if you want to sort by something like date added (oldest first) or custom order across devices.

One workaround I’ve tried: exporting the playlist to a file (like CSV) using a tool such as MusConvtool, then reordering it externally and re‑importing it. It’s a bit manual but helps if you want more advanced sorting than Spotify offers natively.

Would be great if Spotify added more flexible sorting (like multi‑column sorting or pinning certain tracks). Fingers crossed they improve this in future updates!

**Comment by OzGav** [comment]
Playing a playlist in displayed order was added in 2.9.0 beta 7
